src/components/application_manager/src/commands/mobile/subscribe_way_points_request.cc

```diff
@@ -47,14 +48,23 @@ void SubscribeWayPointsRequest::on_event(const event_engine::Event& event) {
   switch (event.id()) {
     case hmi_apis::FunctionID::Navigation_SubscribeWayPoints: {
       LOG4CXX_INFO(logger_, "Received Navigation_SubscribeWayPoints event");
-      mobile_apis::Result::eType result_code =
-          GetMobileResultCode(static_cast<hmi_apis::Common_Result::eType>(
-              message[strings::params][hmi_response::code].asUInt()));
-      bool result = mobile_apis::Result::SUCCESS == result_code;
+      hmi_apis::Common_Result::eType result_code =
+          static_cast<hmi_apis::Common_Result::eType>(
+              message[strings::params][hmi_response::code].asUInt());
```

Comment:
@dtrunov Please replace `asUInt()` with `asInt()`. `hmi_apis::Common_Result::eType` has negative values. Also `result_code` may be constant.